app/src/Services/WorkflowEngine/DefaultWorkflowEngine.php

```
@@ -0,0 +1,838 @@
+
+        return new ServiceResult(true, null, $instance);
+    }
+
+    /**
+     * Attempt a named transition on a workflow instance.
+     */
+    public function transition(int $instanceId, string $transitionSlug, ?int $triggeredBy = null, array $context = []): ServiceResult
+    {
+        $instancesTable = TableRegistry::getTableLocator()->get('WorkflowInstances');
+        $transitionsTable = TableRegistry::getTableLocator()->get('WorkflowTransitions');
+        $statesTable = TableRegistry::getTableLocator()->get('WorkflowStates');
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$statesTable'. (undefined)

(UnusedLocalVariable)